dev-lang/vala/vala-0.36.12.recipe

```diff
@@ -0,0 +1,114 @@
+SUMMARY="An object-oriented programming language that uses the GObject system"
+DESCRIPTION="Vala is a new programming language that aims to bring modern \
+programming language features to GNOME developers without imposing any \
+additional runtime requirements and without using a different ABI \
+compared to applications and libraries written in C."
+HOMEPAGE="https://wiki.gnome.org/Projects/Vala"
+COPYRIGHT="2006-2018 Jürg Billeter
+	2006-2018 Raffaele Sandrini"
+LICENSE="GNU LGPL v2.1"
+REVISION="1"
+SOURCE_URI="https://download.gnome.org/sources/vala/${portVersion%.*}/vala-$portVersion.tar.xz"
+CHECKSUM_SHA256="b4081d3d313cf9c3e8d7d795983cf8f7ed99286d1abc1b5709b0f17883387cdb"
+SOURCE_DIR="vala-$portVersion"
+PATCHES="vala-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion="0.0.0"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+portVersionCompat="$portVersion compat >= ${portVersion%%.*}"
+portVers="${portVersion%.*}"
+
+PROVIDES="
+	vala$secondaryArchSuffix = $portVersionCompat
+	cmd:vala$secondaryArchSuffix
+	cmd:vala_$portVers$secondaryArchSuffix
+	cmd:vala_gen_introspect$secondaryArchSuffix
+	cmd:vala_gen_introspect_$portVers$secondaryArchSuffix
+	cmd:valac$secondaryArchSuffix
+	cmd:valac_$portVers$secondaryArchSuffix
+	cmd:vapicheck$secondaryArchSuffix
+	cmd:vapicheck_$portVers$secondaryArchSuffix
+	cmd:vapigen$secondaryArchSuffix
+	cmd:vapigen_$portVers$secondaryArchSuffix
+	lib:libvala_$portVers$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libglib_2.0$secondaryArchSuffix
+	lib:libgobject_2.0$secondaryArchSuffix
+	lib:libdbus_1$secondaryArchSuffix
+	lib:libffi$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	"
+REPLACES="
+	vala${secondaryArchSuffix}_common
+	"
+
+PROVIDES_devel="
+	vala${secondaryArchSuffix}_devel = $portVersionCompat
+	"
+REQUIRES_devel="
+	vala$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libglib_2.0$secondaryArchSuffix
+	devel:libgobject_2.0$secondaryArchSuffix
+	devel:libdbus_1$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:autoconf
+	cmd:automake
+	cmd:awk
+	cmd:bison
+	cmd:cmp
+	cmd:diff
+	cmd:flex
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+defineDebugInfoPackage vala$secondaryArchSuffix \
+	$binDir/vala \
+	$binDir/valac \
+	$binDir/vapicheck \
+	$binDir/vapigen \
+	$binDir/vala-$portVers \
+	$binDir/valac-$portVers \
+	$binDir/vapicheck-$portVers \
+	$binDir/vapigen-$portVers
```

Comment:
Oops, you ate the lib ;) That's my fault, sorry...
```diff
-	$binDir/vapigen-$portVers
-	$binDir/vapigen-$portVers \
+	$libDir/libvala-$portVers.so.$libVersion
```

Reply:
BTW, what do you think of this variant?
```
 defineDebugInfoPackage vala$secondaryArchSuffix \
 	$binDir/vala \
	$binDir/vala-$portVers \
 	$binDir/valac \
	$binDir/valac-$portVers \
 	$binDir/vapicheck \
	$binDir/vapicheck-$portVers \
 	$binDir/vapigen \
	$binDir/vapigen-$portVers \
	$libDir/libvala-$portVers.so.$libVersion
```
(I've just move some lines so that similar lines together.)